lib/client.js

```diff
@@ -83,13 +83,12 @@ Client.prototype.connect = function () {
             self.emit('brokersChanged');
         }, 3000);
     });
-    zk.once('disconnected', function () {
+    zk.once('expired', function () {
+        // We were disconnected for too long and session expired - recreate the client
         if (!zk.closed) {
             zk.close();
-            setTimeout(function () {
-                self.connect();
-                self.emit('zkReconnect');
-            }, Math.floor(Math.random() * 1000) + 1000);
+            self.connect();
```

Comment:
@d00rman no, no chance to have the same error. The 'expired' event will occur only when ZK is back online and we were able to connect to it successfully. What are the odds that it will die again for a long time right after it got back?